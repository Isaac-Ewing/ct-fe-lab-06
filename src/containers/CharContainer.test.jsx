import React from 'react';
import { screen, render, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
// import charDetails from '../fixtures/CharContainerRes.json';
import ACCharacterDetails from './CharContainer';

const mockServer = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers/5f5fb4bbbfd05c2aed82e460', (req, res, ctx) => {
    return res(ctx.json({
      name: 'Admiral',
      quote: 'Only quitters give up.',
      personality: 'cranky'
    }));
  })
);

describe('chartater detail page', () => {
  beforeAll(() => mockServer.listen());
  afterAll(() => mockServer.close());

  it('has the details of only a single character', async () => {
    render(
      <MemoryRouter initialEntries={[{ pathname: 'https://ac-vill.herokuapp.com/villagers/5f5fb4bbbfd05c2aed82e460' }]}>
        <ACCharacterDetails />
      </MemoryRouter>
    );
    screen.getByAltText('spin');

    return waitFor(() => {
      const character = screen.findByText('Admiral', { exact: false });
      expect(character).toMatchSnapshot();
    }, { timeout: 2000 });
  });
});
