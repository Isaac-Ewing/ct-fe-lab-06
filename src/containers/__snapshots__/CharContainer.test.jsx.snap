// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chartater detail page has the details of only a single character 1`] = `Promise {}`;
